config.go

```diff
@@ -0,0 +1,324 @@
+	Modems: []modem{
+		modem{Name: "Huawei 4G modem", NetDev: "eth1", VendorProductID: "12d1:14db"},
+		modem{Name: "Spark 3G modem", NetDev: "usb0", VendorProductID: "19d2:1405"},
+	},
+}
+
+type Lepton struct {
+	SPISpeed    int64  `mapstructure:"spi-speed"`
+	FrameOutput string `mapstructure:"frame-output"`
+}
+
+var DefaultLepton = Lepton{
+	SPISpeed:    2000000,
+	FrameOutput: "/var/run/lepton-frames",
+}
+
+type ThermalRecorder struct {
+	OutputDir      string `mapstructure:"output-dir"`
+	MinDiskSpaceMB uint64 `mapstructure:"min-disk-space-mb"`
+	MinSecs        int    `mapstructure:"min-secs"`
+	MaxSecs        int    `mapstructure:"max-secs"`
+	PreviewSecs    int    `mapstructure:"preview-secs"`
+}
+
+var DefaultThermalRecorder = ThermalRecorder{
+	MaxSecs:        600,
+	MinSecs:        10,
+	PreviewSecs:    3,
+	MinDiskSpaceMB: 200,
+	OutputDir:      "/var/spool/cptv",
+}
+
+type ThermalMotion struct {
+	DynamicThreshold bool   `mapstructure:"min-secs"`
+	TempThresh       uint16 `mapstructure:"temp-thresh"`
+	DeltaThresh      uint16 `mapstructure:"delta-thresh"`
+	CountThresh      int    `mapstructure:"count-thresh"`
+	FrameCompareGap  int    `mapstructure:"frame-compare-gap"`
+	UseOneDiffOnly   bool   `mapstructure:"use-one-diff-only"`
+	TriggerFrames    int    `mapstructure:"trigger-frames"`
+	WarmerOnly       bool   `mapstructure:"warmer-only"`
+	EdgePixels       int    `mapstructure:"edge-pixels"`
+	Verbose          bool   `mapstructure:"verbose"`
+}
+
+var DefaultThermalMotion = ThermalMotion{
+	DynamicThreshold: true,
+	TempThresh:       2900,
+	DeltaThresh:      50,
+	CountThresh:      3,
+	FrameCompareGap:  45,
+	Verbose:          false,
+	TriggerFrames:    2,
+	UseOneDiffOnly:   true,
+	WarmerOnly:       true,
+	EdgePixels:       1,
+}
+
+type ThermalThrottler struct {
+	Activate   bool
+	BucketSize time.Duration `mapstructure:"bucket-size"`
+	MinRefill  time.Duration `mapstructure:"min-refill"`
+}
+
+var DefaultThermalThrottler = ThermalThrottler{
+	Activate:   true,
+	BucketSize: 10 * time.Minute,
+	MinRefill:  10 * time.Minute,
+}
+
+type Ports struct {
+	Managementd int
+}
+
+var DefaultPorts = Ports{
+	Managementd: 80,
+}
+
+type Secrets struct {
+	DevicePassword string `mapstructure:"device-password"`
+}
+
+type GPIO struct {
+	ThermalCameraPower string `mapstructure:"thermal-camera-power"`
+	ModemPower         string `mapstructure:"modem-power"`
+}
+
+var DefaultGPIO = GPIO{
+	ThermalCameraPower: "GPIO23",
+	ModemPower:         "GPIO22",
+}
+
+const (
+	DeviceKey           = "device"
+	WindowsKey          = "windows"
+	LocationKey         = "location"
+	TestHostsKey        = "test-hosts"
+	BatteryKey          = "battery"
+	ModemdKey           = "modemd"
+	LeptonKey           = "lepton"
+	ThermalRecorderKey  = "thermal-recorder"
+	ThermalMotionKey    = "thermal-motion"
+	ThermalThrottlerKey = "thermal-throttler"
+	PortsKey            = "ports"
+	SecretsKey          = "secrets"
+	GPIOKey             = "gpio"
```

Comment:
To aid discoverability and maintenance (i.e. less change of merge conflicts) I would split the above out it's their own files - one per config section (so device.go, ports.go, gpio.go) etc. The relevant key constant for each could live at the top of each file too.